Guidance, Part 3: Help

rod-glover opened the pull request:
Issue #173 Guidance
Part 3: Help, etc. (cherry picked from humongous PR #174)

Fill in help and related content, namely content of all the following navigation items. This is distinct from info popovers.

- Help > *
- Science
- About > *

corviday commented on src/components/guidance-content/help/HelpGeneral.js:
```diff
@@ -0,0 +1,724 @@
+                    precipitation
+                  </p>
+                  <p>
+                    This is a variant ('alt') version of the <a href='https://www.climdex.org/indices.html'>Climdex</a> variable <em>CDD</em>.
```
"This is an annual version of the monthly Climdex variable cwd"

corviday commented on src/components/guidance-content/help/HelpGeneral.js:
```diff
@@ -0,0 +1,724 @@
+                </ListGroupItem>
+                <ListGroupItem header='altwsdiETCCDI'>
+                  <p>
+                    Warm spell duration index spanning years
```
Why do we even have this variable? There isn't gonna be a warm spell from December to January, is there?
Anyway.

"This is an alternative version of the Climdex variable csdi that counts the full length of cold spells that extend from December of the target yeaar through January of the next year."

rod-glover replied:
Not in the northern hemisphere ... at least not until global warming gets much worse.

corviday commented on src/components/guidance-content/help/Glossary.js:
```diff
@@ -0,0 +1,60 @@
+
+
+// Items are sorted automatically here.
+
```
I made a list of possibly unfamiliar terms from text in the app. I don't know if we need these in the glossary, since they're all explained in the popovers.
- model
- emissions scenario
- variable
- dataset
- xlsx
- csv
- anomaly (if we keep using that name - one of my PRs renames it.)
- model run
- raster
- isoline
- layer

And two acronyms I didn't see any explanation for:
- GCM
- CMIP5

In summary, I'm not sureif we really need a glossary or what should go in it. Most things are already explained somewhere.

rod-glover replied:
Removed Glossary for now.

corviday commented on src/components/guidance-content/help/HelpGeneral.html:
```diff
@@ -0,0 +1,330 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
+<html>
```
This page is huge. Can it be given a table of contents, ideally with links? I vaguely recall you saying something about anchor links not working, though, so maybe not.

rod-glover replied:
Love to, but time...

rod-glover replied:
https://github.com/pacificclimate/climate-explorer-frontend/issues/206

corviday commented:
>     1. Help. Ordering of subitems (maybe General first, then FAQ, as we have no FAQ content)?

I agree General first would be good.

>     2. Help > Glossary. Any suggestions for content here?

Answered inline.

>     3. About > Team. Too effusive, unserious?

Seems okay to me.

corviday commented on src/components/guidance-content/help/HelpGeneral.js:
```diff
@@ -0,0 +1,724 @@
+          <p>It helps to have a little terminology for the various parts of the
+            Climate Explorer application.</p>
+          <p>
+            <img src={overviewImg} alt='Overview of app'/>
```
If not the whole image, at least the text.

rod-glover replied:
@corviday , let me know when this is done and I will integrate it. I'd like it to be a full-app image, which you may have to assemble from 2 partial ones (scrolled to different points).

corviday replied:
How big would you like the image to be?

rod-glover commented on src/components/guidance-content/help/HelpGeneral.js:
```diff
@@ -0,0 +1,724 @@
+              precipitation events that would be expected to occur once every
+              twenty years. T
+              his data is available for six thirty year periods from
+              1960 to 2100.
```
Those periods being?

corviday replied:
1971-2000
2011-2040
2041-2070
2071-2100

rod-glover replied:
done

rod-glover commented on src/components/guidance-content/help/HelpGeneral.js:
```diff
@@ -0,0 +1,724 @@
+                    Cold spell duration index spanning years
+                  </p>
+                  <p>
+                    This is a variant ('alt') version of the <a href='https://www.climdex.org/indices.html'>Climdex</a> variable <em>CSDI</em>.
```
We have a bigger problem here: 

Climdex defines CDD as "Maximum length of dry spell, maximum number of consecutive days with RR < 1mm".

Elsewhere in this help, and you are proposing here also, we define the variable cdd as meaning "cooling degree days".

We need to (a) find out which CDD this one is, and (b) whether we really do compute cooling degree days and whether there is a conflict of variable naming here.

rod-glover replied:
Oops, I thought this was attached to CDD, not CSDI. But we still have the problem that CDD means two different things.

rod-glover replied:
Hmm, I guess our variable names do distinguish the two cases: 'cdd' for cooling degree days, and 'cddETCCDI' for dry spell. I will make the appropriate clarifications.

rod-glover commented on src/components/guidance-content/help/HelpGeneral.js:
```diff
@@ -0,0 +1,724 @@
+              (although seasonal and monthly degree-day counts are also
+              possible).
+              The data is averaged over several thirty year periods between
+              1970 to 2100.
```
Those periods being?

corviday replied:
Sorry, missed this earlier. 

1971-2000
2011-2040
2041-2070
2071-2100

rod-glover replied:
done

rod-glover commented on src/components/guidance-content/help/HelpGeneral.js:
```diff
@@ -0,0 +1,724 @@
+            datasets. To convey the maximum amount of information to the human
+            user, the minimum and maximum values should be close to the dataset’s
+            minimum and maximum values.</p>
+          <p>We offer two different types of color scale, linear and
```
You are definitely culturally aware of our unique Canuckness!

rod-glover commented on src/components/guidance-content/help/FAQ.js:
```diff
@@ -0,0 +1,95 @@
+
+
+const faqs = [
+  {
```
done

rod-glover commented on src/components/guidance-content/help/HelpGeneral.js:
```diff
@@ -0,0 +1,724 @@
+              </p>
+              <h5>Variable names and meanings</h5>
+              <ListGroup>
+              </ListGroup>
```
done

rod-glover commented on src/components/guidance-content/help/HelpGeneral.js:
```diff
@@ -0,0 +1,724 @@
+import React from 'react';
```
done

rod-glover commented:
@corviday , thank you for the thorough review! This addresses all the concerns you identified (in some cases by not doing anything or deferring to a separate issue to be dealt with later--so noted). There is one element I am not yet happy about, which is a clear statement of data provenance and type (e.g., BCCAQ v2 vs. non-downscaled datasets, if any). I've put this in issue https://github.com/pacificclimate/climate-explorer-frontend/issues/207.

I'd like to get this merged ASAP, so if I could get your response to this quickly that would be great.

corviday commented:
Do you need the new image too, or is the review higher priority?

rod-glover commented:
@corviday , sorry for the delay. (ironic)

We'll leave the image for later.

New [demo](http://docker-dev02.pcic.uvic.ca:30503). up now.

Kindly ignore the latest commit (with duplicate message, argh), which will be removed shortly. It is a fake to force Dockerhub to do a build. Somehow a glitch happened and it missed the previous commit pushes today.

rod-glover commented:
Fake commit removed
